File: CSharp/Parsers/Common/NodeManager.cs
using Parsers.Common;
using System.Collections.Generic;

namespace Parsers.Common
{
    public class NodeManager {
        public List<object> nodes;
        public Reader reader;
        
        public NodeManager(Reader reader)
        {
            this.reader = reader;
            this.nodes = new List<object>();
        }
        
        public void addNode(object node, int start)
        {
            this.nodes.push(node);
        }
    }
}

File: PHP/Parsers/Common/NodeManager.php
<?php

namespace Parsers\Common\NodeManager;

use Parsers\Common\Reader\Reader;

class NodeManager {
    public $nodes;
    public $reader;
    
    function __construct($reader) {
        $this->reader = $reader;
        $this->nodes = array();
    }
    
    function addNode($node, $start) {
        $this->nodes[] = $node;
    }
}
